tests/e2e/main.go

```diff
@@ -33,6 +27,23 @@ func (t *TestSet) String() string {
 	return fmt.Sprint(*t)
 }
 
+type VersionSet []string
+
+func (c *VersionSet) Set(value string) error {
+	// Check and skip duplicates
+	for _, v := range *c {
+		if v == value {
+			return nil
+		}
+	}
+	*c = append(*c, value)
+	return nil
+}
+
+func (t *VersionSet) String() string {
```

Comment:
nit: perhaps `vs` would be easier to understand than `t` as a short name for `VersionSet`.